### PR TITLE
Allow us to define what kind of machines and node count for GKE

### PR DESCRIPTION
**Pull request by aaomidi**
We're spinning up a LOT of nodes and big machines for dev environments and staging, we don't need to do this. This PR surfaces the variables used when constructing the GKE environment.

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -172,6 +167,14 @@ variable "victorops_routing_key" {
   description = "VictorOps/Splunk OnCall routing key for prometheus-alertmanager"
   default     = "bogus-routing-key"
 }
+variable "cluster_settings" {
+  type = object({
+    initial_node_count : number
```
Comment (tgeoghegan) [inline terraform/main.tf:172]: I didn't know this was also valid syntax for TF object definitions! Elsewhere we use `name = type`, and if possible I'd prefer to use that here for consistency.

File: terraform/variables/prod-us.tfvars [hunk 2]
```diff
@@ -69,6 +68,12 @@ ingestors = {
     }
   }
 }
+cluster_settings = {
+  initial_node_count : 4
+  min_node_count : 4
+  max_node_count : 5
```
Comment (tgeoghegan) [inline terraform/variables/prod-us.tfvars:74]: As elsewhere, please use `label = value` style for consistency.

Comment (tgeoghegan) [inline terraform/variables/prod-us.tfvars:74]: The only reason that `initial_node_count` and `min_node_count` were 4 was to hack around a GKE-imposed resource limit on the number of Kubernetes jobs per namespace, which was necessary back when we spawned one k8s job per task. We don't need to do that anymore, so we can use lower values to allow the cluster to scale down if nodes are idle. Let's try `initial_node_count = 1` and `min_node_count = 3`.

File: terraform/variables/staging-facil.tfvars [hunk 3]
```diff
@@ -45,6 +44,12 @@ ingestors = {
     }
   }
 }
+cluster_settings = {
+  initial_node_count : 2
+  min_node_count : 2
+  max_node_count : 3
+  machine_type : "e2-standard-4"
```
Comment (tgeoghegan) [inline terraform/variables/staging-facil.tfvars:51]: Let's try `e2-standard-2`.

File: terraform/variables/staging-facil.tfvars [hunk 4]
```diff
@@ -45,6 +44,12 @@ ingestors = {
     }
   }
 }
+cluster_settings = {
+  initial_node_count : 2
+  min_node_count : 2
```
Comment (tgeoghegan) [inline terraform/variables/staging-facil.tfvars:49]: Our node pools are regional/multi-zonal, so all these values are per-AZ. I think one node per AZ will be enough for staging. Can we try `initial_node_count = 1` and `min_node_count = 1`?

**Review by tgeoghegan — CHANGES_REQUESTED** [review]

File: terraform/main.tf (hunk 1 above)
Comment (aaomidi) [reply terraform/main.tf:172]: I actually got this syntax for where we define ingestors: https://github.com/abetterinternet/prio-server/blob/main/terraform/main.tf#L36-L45

Comment (aaomidi) [reply terraform/main.tf:172]: Oh now I see what you meant, using `=` instead of `:`

I'm really curious if there is a difference between these two.

**Pull request merged**